lib/facter/udevadm.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# Fact: udevadm
+#
+# Purpose:
+#   Provide information about the udevadm utilty.
+#
+# Resolution:
+#   Check if the kernel is linux.
+#
+# Caveats:
+#
+Facter.add(:udevadm) do
+  confine kernel: :linux
```

Comment:
I think a bit cleaner is it to confine onto the existence of `udevadm`:
```
confine { Facter::Core::Execution.which('udevadm') }
```

then we don't need to wrap the setcode into an if block

Reply:
OK - I think this is a reasonable approach.  A possible downside is that it doesn't appear that there is any memoization on top of `Facter::Core::Execution::Posix.which` but odds are that inode data for the `stat` calls will be cached by the OS.

Reply:
You can just move everything inside `setcode` and make it return `nil` if `path` is nil. So something like this probably works:
```ruby
setcode do
  path = Facter::Util::Resolution.which('udevadm')
  if path
    {}
  end
end

Reply:
Done.

Reply:
Overlapping comments, @ekohl last suggestion hadn't appear when I marked this resolved.  I like that approach even better.  Standby...

Reply:
[re]done.